packages/core/src/node/mdx/loader.ts

```diff
@@ -88,41 +90,44 @@ export default async function mdxLoader(
   source: string,
   callback: Rspack.LoaderContext['callback'],
 ) {
+  context.cacheable(true);
+
   const options = context.getOptions();
   const filepath = context.resourcePath;
   const { alias } = context._compiler.options.resolve;
-  context.cacheable(true);
-  let pageMeta = {
-    title: '',
-    toc: [],
-  } as PageMeta;
-
   const { config, docDirectory, checkDeadLinks, routeService, pluginDriver } =
     options;
 
+  // Separate frontmatter and content in MDX source
   const { frontmatter, content } = loadFrontMatter(
     source,
     filepath,
     docDirectory,
     true,
   );
 
+  // Replace imported built-in MDX content
   const { flattenContent, deps } = await flattenMdxContent(
     content,
     filepath,
     alias as Record<string, string>,
   );
-  // replace content
+
+  deps.forEach(dep => context.addDependency(dep));
+
+  // Resolve side effects caused by flattenMdxContent.
+  // Perhaps this problem should be solved within flattenMdxContent?
```

Comment:
what's this mean?

Reply:
`FlattenMdxContent` can cause the unfolded MDX content to not apply `replaceRules`. To solve this problem, I think `replaceRules` should be applied to each MDX content flatten in the `flattenMdxContent`.

Reply:
so it's a TODO?

Reply:
If you agree, I will modify the logic of `flattenMdxContent` afterwards.

Reply:
open a issue first and not in this pr